src/docfx/build/context/BuildScope.cs

```diff
@@ -251,6 +252,17 @@ private static (Func<string, bool>, FileMappingConfig)[] CreateGlobs(Config conf
                         return JsonUtility.ReadMime(reader, filePath);
                     }
                 case FileFormat.Yaml:
+                    if (filePath.Origin == FileOrigin.Generated)
+                    {
+                        var servicePageToken = input.ReadYaml(errors, filePath).SelectToken("yamlMime");
```

Comment:
Add an additional `mime` parameter to `AddGeneratedContent` method to pass mime type over using this special `yamlMime` property name.

Reply:
Add `yamlMime` pamerater to generatedContent as `private readonly ConcurrentDictionary<FilePath, (string? yamlMime, JToken generatedContent)> _generatedContents = new ConcurrentDictionary<FilePath, (string?, JToken)>();`